spec.md

```diff
@@ -15,4 +16,574 @@ The key words "unspecified", "undefined", and "implementation-defined" are to be
+
+* Specific mechanisms by which a Plugin Supervisor manages the lifecycle of a Plugin, including:
+  * How to deploy, install, upgrade, uninstall, monitor, or respawn (in case of unexpected termination) Plugins.
+* A first class message structure/field to represent "grades of object storage" (aka "bucket class").
+* Protocol-level authentication and authorization.
+* Packaging of a Plugin.
+
+## Solution Overview
+
+This specification defines an interface along with the minimum operational and packaging recommendations for a storage provider (SP) to implement a COSI compatible plugin.
+The interface declares the RPCs that a plugin MUST expose: this is the **primary focus** of the COSI specification.
+Any operational and packaging recommendations offer additional guidance to promote cross-COSI system compatibility.
+
+### Architecture
+
+The primary focus of this specification is on the **protocol** between a COSI System and a Plugin.
+It SHOULD be possible to ship cross-COSI system compatible Plugins for a variety of deployment architectures.
+A COSI system SHOULD be equipped to handle both centralized and headless plugins.
+Several of these possibilities are illustrated in the following figures.
+
+```
+                    COSI System "Master" Host
++-------------------------------------------+
+|                                           |
+|  +------------+           +------------+  |
+|  |   COSI     |   gRPC    | Controller |  |
+|  |   System   +----------->   Plugin   |  |
+|  +------------+           +------------+  |
+|                                           |
++-------------------------------------------+
+
+```
+Figure 1: Plugin deployment, all of the COSI system Node hosts run Plugins. 
+
+
+![Cosi Components](cosicomponents.png)
+Figure 2: Interaction within COSI components 
+
+### Bucket Lifecycle
+
+```
+     CreateBucket +-------------+ DeleteBucket 
+    +------------->|  CREATED   +--------------+ 
+    |              +---+----^---+              |    
+    |          Grant   |    | Revoke           v    
+    +++        Bucket  |    | Bucket          +++  
+    |X|        Access  |    | Access          | |  
+    +-+            +---v----+---+             +-+  
+                   |   BOUND    |                  
+                   +---+----^---+                  
+
+```
+Figure 2: The lifecycle of a dynamically provisioned bucket, from
+creation to destruction.
+```
+
+The above diagrams illustrate a general expectation with respect to how a COSI system MAY manage the lifecycle of a bucket via the API presented in this specification. 
+Plugins SHOULD expose all RPCs for an interface: Controller plugins SHOULD implement all RPCs for the  service.
+Unsupported RPCs SHOULD return an appropriate error code that indicates such (e.g. `CALL_NOT_IMPLEMENTED`).
+
+## Container Object Storage Interface
+
+This section describes the interface between COSI systems and Plugins.
+
+### RPC Interface
```

Comment:
Does this section contain any changes that are still being reviewed in this other PR https://github.com/kubernetes-sigs/container-object-storage-interface-spec/pull/20?

Reply:
It does not. I will update this section in a separate PR.